File: examples/php/ds/engine.php
<?php
namespace books;
require_once __DIR__ . '/../common/data/base.php';
require_once __DIR__ . '/api/author.php';
require_once __DIR__ . '/api/book.php';
require_once __DIR__ . '/api/genre.php';
require_once __DIR__ . '/api/publisher.php';


class Engine {
    public $db = null;
    public $authors = null;
    public $books = null;
    public $genres = null;
    public $publishers = null;

    function __construct(string $scheme, ?string $queries=null) {
        $this->db = new \quartz\Database($scheme, $queries);
        $this->authors = new AuthorManager($this->db, $this);
        $this->books = new BookManager($this->db, $this);
        $this->genres = new GenreManager($this->db, $this);
        $this->publishers = new PublisherManager($this->db, $this);
    }

    function open(bool $reset=false) {
        $this->db->open(getenv('DB_NAME'), getenv('DB_USER'), getenv('DB_PASS'), reset:$reset);
    }
}

?>

File: examples/python/ds/engine.py
import ds.api._core as _core
from ds.api.author import *
from ds.api.book import *
from ds.api.genre import *
from ds.api.publisher import *
import quartz.db.sqlite as _sqlite

#
# Engine
#

class Engine(_core.Engine):

    def __init__(self, schema_path:str, file_path:str, reset:bool=False):
        db = _sqlite.Database(schema_path, file_path)
        db.open(reset)
        # Entities
        self._authors = AuthorManager(db, self)
        self._books = BookManager(db, self)
        self._genres = GenreManager(db, self)
        self._publishers = PublisherManager(db, self)

